node/worker/worker.go

```diff
@@ -461,11 +462,8 @@ func (w *Worker) FinalizeNewBlock(
 		}
 	}
 
-	state := w.current.state.Copy()
-	copyHeader := types.CopyHeader(w.current.header)
-	// TODO: feed coinbase into here so the proposer gets extra rewards.
 	block, _, err := w.engine.Finalize(
-		w.chain, copyHeader, state, w.current.txs, w.current.receipts,
```

Comment:
You made me add the header copy , now taking it out lol

Reply:
Yea, at first it's for safety. But after looking deeper, now can remove for speed.